src/handlers.rs

```diff
@@ -979,6 +979,92 @@ pub fn get_banned_public_keys(
+    }
+    if !is_moderator(&requesting_public_key, pool)? {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    // Get a connection and open a transaction
+    let mut conn = pool.get().map_err(|_| Error::DatabaseFailedInternally)?;
+    let tx = conn.transaction().map_err(|_| Error::DatabaseFailedInternally)?;
+    // Pin the message if it's present
+    let stmt = format!("UPDATE MAIN_TABLE SET pinned_message_id = (?1)");
+    match tx.execute(&stmt, params![id]) {
+        Ok(count) => count,
+        Err(e) => {
+            error!("Couldn't pin message due to error: {}.", e);
+            return Err(warp::reject::custom(Error::DatabaseFailedInternally));
+        }
+    };
+    // Commit
+    tx.commit().map_err(|_| Error::DatabaseFailedInternally)?;
+    // Return
+    let json = models::StatusCode { status_code: StatusCode::OK.as_u16() };
+    return Ok(warp::reply::json(&json).into_response());
+}
+
+pub fn get_pinned_message(auth_token: &str, pool: &storage::DatabaseConnectionPool) -> Result<i64, Rejection> {
+    // Check authorization level
+    let (has_authorization_level, _) = has_authorization_level(auth_token, AuthorizationLevel::Basic, pool)?;
+    if !has_authorization_level {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    // Get a database connection
+    let conn = pool.get().map_err(|_| Error::DatabaseFailedInternally)?;
+    // Query the database
+    let raw_query = format!("SELECT pinned_message_id FROM MAIN_TABLE");
+    let mut query = conn.prepare(&raw_query).map_err(|_| Error::DatabaseFailedInternally)?;
+    let rows = match query.query_map(params![], |row| row.get(0)) {
+        Ok(rows) => rows,
+        Err(e) => {
+            error!("Couldn't query database due to error: {}.", e);
+            return Err(warp::reject::custom(Error::DatabaseFailedInternally));
+        }
+    };
+    let pinned_message: i64 = rows
+        .filter_map(|result| result.ok())
+        .next()
+        .ok_or_else(|| warp::reject::custom(Error::DatabaseFailedInternally))?;
+    return Ok(pinned_message);
+}
+
+pub fn delete_pinned_message(auth_token: &str, pool: &storage::DatabaseConnectionPool) -> Result<Response, Rejection> {
+    // Check authorization level
+    let (has_authorization_level, requesting_public_key) =
+        has_authorization_level(auth_token, AuthorizationLevel::Basic, pool)?;
+    if !has_authorization_level {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    if !is_moderator(&requesting_public_key, pool)? {
+        return Err(warp::reject::custom(Error::Unauthorized));
+    }
+    // Get a connection and open a transaction
+    let mut conn = pool.get().map_err(|_| Error::DatabaseFailedInternally)?;
+    let tx = conn.transaction().map_err(|_| Error::DatabaseFailedInternally)?;
+    // Pin the message if it's present
+    let stmt = format!("UPDATE MAIN_TABLE SET pinned_message_id = (?1)");
+    match tx.execute(&stmt, [0]) {
```

Comment:
Does setting to 0 cause issues? The ID for the messages table is automatically generated by sqlite

Reply:
Jubb:
it'll be just easier on our end I think on deserialising the json object to check if pinned message exists
rather than treat a specific number as a different case
so I was gonna say it might make sense to represent it internally as -1 or some other null value that is more suited (maybe jason has some idea) and then if it IS that value, just don't include it in the serialized json object, as null or w/e
then on our end we can just go "does pinned message exist" and branch on that
otherwise treat its absence in the json object as being "there is no pinned message"

Reply:
The only thing that could potentially conflict with is if the pinned message is exposed as a separate endpoint, what value, if any, do we return there

Reply:
> so I was gonna say it might make sense to represent it internally as -1 or some other null value that is more suited (maybe jason has some idea) and then if it IS that value, just don't include it in the serialized json object, as null or w/e
then on our end we can just go "does pinned message exist" and branch on that

The column added is nullable, so I'd just set it to `NULL` and chase that through.  (In terms of the JSON API: you *could* set it to `null`, but I think it'd be slightly cleaner to just omit the key entirely when null).